twilio/config-types.bal

```diff
@@ -0,0 +1,489 @@
+    decimal timeout = 60;
+    # The choice of setting `forwarded`/`x-forwarded` header
+    string forwarded = "disable";
+    # Configurations associated with request pooling
+    PoolConfiguration poolConfig?;
+    # HTTP caching related configurations
+    CacheConfig cache = {};
+    # Specifies the way of handling compression (`accept-encoding`) header
+    Compression compression = COMPRESSION_AUTO;
+    # Configurations associated with the behaviour of the Circuit Breaker
+    CircuitBreakerConfig circuitBreaker?;
+    # Configurations associated with retrying
+    RetryConfig retryConfig?;
+    # Configurations associated with inbound response size limits
+    ResponseLimitConfigs responseLimits = {};
+    #SSL/TLS-related options
+    ClientSecureSocket secureSocket?;
+    # Proxy server related options
+    ProxyConfig proxy?;
+    # Enables the inbound payload validation functionalty which provided by the constraint package. Enabled by default
+    boolean validation = true;
+|};
+
+# Twilio Token Based Authentication
+#
+# + accountSId - Twilio account SID  
+# + authToken - The authentication token of the account 
+@display{label: "Auth Token Based Connection Config"} 
+public type TokenBasedAuthentication record {
+    @display{label: "Account SID"} 
+    string accountSId;
+    @display{label: "Auth Token"} 
+    string authToken;
+};
+
+# Twilio API Key Based Authentication
+#
+# + accountSId - Twilio account SID  
+# + apiKey - Twilio API key SID 
+# + apiSecret - Twilio API key Secret
+@display{label: "API Key Based Connection Config"} 
+public type APIKeyBasedAuthentication record {
+    @display{label: "Twilio Account SID"} 
+    string accountSId;
+    @display{label: "API Key SID"} 
+    string apiKey;
+    @display{label: "API Key Secret"} 
+    string apiSecret;
+};
+
+# Represents token for Bearer token authentication.
+#
+# + token - Bearer token for authentication
+public type BearerTokenConfig record {|
+    string token;
+|};
+
+# Represents OAuth2 refresh token grant configurations for OAuth2 authentication.
+#
+# + refreshUrl - Refresh token URL of the token endpoint
+# + refreshToken - Refresh token for the token endpoint
+# + clientId - Client ID of the client authentication
+# + clientSecret - Client secret of the client authentication
+# + scopes - Scope(s) of the access request
+# + defaultTokenExpTime - Expiration time (in seconds) of the tokens if the token endpoint response does not contain an `expires_in` field
+# + clockSkew - Clock skew (in seconds) that can be used to avoid token validation failures due to clock synchronization problems
+# + optionalParams - Map of the optional parameters used for the token endpoint
+# + credentialBearer - Bearer of the authentication credentials, which is sent to the token endpoint
+# + clientConfig - HTTP client configurations, which are used to call the token endpoint
+public type OAuth2RefreshTokenGrantConfig record {|
```

Comment:
As these are not used, we dont need auth related records.
Please refer to https://github.com/ballerina-platform/module-ballerinax-peoplehr/pull/13